extensions/job_management/standard/sections/clause_6_job_management.adoc

```diff
@@ -0,0 +1,257 @@
+
+
+[[job-management-update]]
+=== Updating an existing job
+
+==== Sequence diagram
+
+The following diagram illustrates the sequence diagram for updating a
+previously created job.  The identifier of the job does not change.
+
+NOTE: The new job definition replaces the old job definition. Version control is not discussed in this Standard.
+
+```
+Client                                                        Server
+  |                                                             |
+  |  PATCH /jobs/{jobId}   HTTP/1.1                             |
+  |  Content-Type: application/json                             |
+  |                                                             |
+  |------------------------------------------------------------>|
+  |                                                             |
+  |  HTTP/1.1 200 OK                                            |
+  |<------------------------------------------------------------|
+```
+
+==== Operation
+
+include::../requirements/job-management/update/REQ_patch-op.adoc[]
+
+==== Request body
+
+==== Overview
+
+include::../requirements/job-management/update/REQ_body.adoc[]
+
+include::../recommendations/job-management/update/PER_body.adoc[]
+
+include::../requirements/job-management/update/REQ_content-type.adoc[]
+
+include::../recommendations/job-management/update/PER_content-schema.adoc[]
+
+===== OGC API - Processes - Workflow Execute Request
+
+include::../recommendations/job-management/update/REC_body-ogcapi-processes.adoc[]
+
+===== OpenEO Process Graph
+
+include::../recommendations/job-management/update/REC_body-openeo.adoc[]
+
+==== Response
+
+include::../requirements/job-management/update/REQ_response.adoc[]
+
+The status code depends on the server. If the server has replaced the job, the response is either `200` (if the response includes additional content) or `204` (if the response has no additional content).
+
+==== Exceptions
+
+See <<job-management-http_status_codes>> for general guidance.
+
+If the request body's media type is not supported by the server, see requirement /req/deploy-replace-undeploy/deploy/unsupported-media-type from <<OAProc-2>>.
+
+If the job with the specified identifier does not exist on the server, see requirement /req/core/job-exception-no-such-job from <<OAProc-1>>.
+
+include::../requirements/job-management/update/REQ_response-locked.adoc[]
+
+[[job-management-start]]
+=== Staring a job
+
+==== Sequence diagram
+
+The following diagram illustrates the sequence diagram for starting 
+the execution of a previously created jobs.
+
+
+```
+Client                                                        Server
+  |                                                             |
+  |  POST /jobs/{jobId}/results   HTTP/1.1                      |
+  |------------------------------------------------------------>|
+  |                                                             |
+  |  HTTP/1.1 200 OK                                            |
```

Comment:
In openEO we return 202 due to the HTTP status code semantics.

Reply:
In this case, the alignment is with *OGC API - Processes* that defines sync execution by default, which returns the results contents inline, hence the 200. However, 202 could also be returned with a Job Status response if the job was previously configured to run asynchronously.

Things to consider for the resolution here:
- job definition enforcing some execution strategy
- process definition enforcing some execution strategy
- `Prefer` header in `POST /jobs/{jobId}/results` to request an execution strategy

Another important item to consider here is that, `POST /jobs/{jobId}/results` only makes sense if the `POST /jobs` was also negotiated with a `status: created`. This is because `POST /jobs` can also return the 200/201 directly with the sync/async execution immediately triggered. In such case, `POST /jobs/{jobId}/results` should respond with HTTP 423 Locked since the job would already be accepted/running/completed.

It wouldn't hurt to have more examples here to represent each combination.

Reply:
Can the backend decide what the default execution mode is? So there not really any negotiation to get "created", which is because there might be operations that you may want to run before actually starting the computation (e.g. estimating the costs).

In openEO it's certainly async, and for OAP it seems sync (which is a bit counterintuitive for me, but that might be because we call jobs actually batch jobs in openEO and batch jobs in synchronous mode feel wrong).

Reply:
IMO, yes, the backend should be free to decide.

I agree with you about OAP being counter-intuitive of defaulting to sync. It is this way solely for helping implementation support sync-only, which does not require any "job" concept. However, given that Part 4 is all about "job management", I feel implementations should be free to pick their own default. Even in Core, implementation that do support "job" usually implement it because they need async by default...

There should probably be a way to advertise which is the default. In OAP v1.0, there was actually a poorly documented method, which was to order the `jobControlOptions` with the first value being the default execution mode. This allowed per-process defaults.

With Part 4, we must actually consider 3 execution modes when creating jobs:
- sync (technically OAP's default)
  - starts execution right away
  - returns the results inline
- async
  - queued immediately
  - starts whenever resources are available
  - returns a job status response
- create
  - creates the job
  - not started until triggered* (`POST /jobs/{jobId}/results`)
  - returns a job `status: created` response with

(*) triggering can be done as sync or async using `Prefer`, as if submitted on `POST /processes/{processId}/execution`

Therefore, openEO's default is actually `create` followed by `async` "on trigger".
I would like to distinguish this follow-up `async` trigger from the "typical" `async` in OAP. Otherwise, the specification is confusing by reusing the same terms.